### PR TITLE
make VS not crash when service hub failed to launch

### PR DESCRIPTION
**Pull request by heejaechang**
this done so that roslyn with remote host on by default can run in any branch. otherwise, we either can't turn on remote host by default or it only works on certain branch.

### CODE REVIEW

**Comment by heejaechang** [comment]
test vsi please

**Comment by heejaechang** [comment]
retest windows_eta_open_prtest  please

flaky test

**Comment by heejaechang** [comment]
retest windows_vsi_p2_open_prtest please

machine issue

**Comment by heejaechang** [comment]
test vsi please

**Comment by heejaechang** [comment]
retest  windows_eta_open_prtest  please

machine issue. git repo clone failed

**Comment by heejaechang** [comment]
@KevinH-MS @srivatsn @mavasani @panopticoncentral @dotnet/roslyn-ide can you take a look?

made vs to not crash even if service hub is not present. this should let vsi tests to run with remote host on by default.

File: src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostOptions.cs [hunk 1]
```diff
@@ -9,7 +9,7 @@ internal static class RemoteHostOptions
         public const string OptionName = "FeatureManager/Features";
 
         [ExportOption]
-        public static readonly Option<bool> RemoteHost = new Option<bool>(OptionName, nameof(RemoteHost), defaultValue: false);
```
Comment (KevinH-MS) [inline src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostOptions.cs]: Was this change intentional?

Comment (heejaechang) [inline src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostOptions.cs]: yep. @srivatsn want to make remote host on by default.

File: src/VisualStudio/RemoteHostClientMock/Remote/RemoteHostClientFactory.cs [hunk 2]
```diff
@@ -15,13 +15,20 @@ internal class RemoteHostClientFactory : IRemoteHostClientFactory
...
+                {
+                    return InProcRemoteHostClient.CreateAsync(workspace, cancellationToken);
+                }
 
-            return ServiceHubRemoteHostClient.CreateAsync(workspace, cancellationToken);
+                return ServiceHubRemoteHostClient.CreateAsync(workspace, cancellationToken);
+            }
+            catch
+            {
+                return Task.FromResult<RemoteHostClient>(null);
```
Comment (KevinH-MS) [inline src/VisualStudio/RemoteHostClientMock/Remote/RemoteHostClientFactory.cs]: Is this different from ```return SpecializedTasks.Default<RemoteHostClient>();``` (used above)?  If not, make consistent?

File: src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostOptions.cs (hunk 1 above)
Comment (KevinH-MS) [inline src/VisualStudio/Core/Def/Implementation/Remote/RemoteHostOptions.cs]: Ok 👍

**Comment by KevinH-MS** [comment]
LGTM

File: src/VisualStudio/RemoteHostClientMock/Remote/RemoteHostClientFactory.cs (hunk 2 above)
Comment (heejaechang) [inline src/VisualStudio/RemoteHostClientMock/Remote/RemoteHostClientFactory.cs]: specialized tasks is not available in this proj. and I dont want to take more dependency to this csproj than needed due to all kinds of project.json version conflicts caused by .Next project

**Comment by heejaechang** [comment]
test vsi please

**Comment by heejaechang** [comment]
retest  windows_vsi_p0_open_prtest please

**Comment by heejaechang** [comment]
retest windows_vsi_p3_open_prtest please

**Comment by heejaechang** [comment]
retest windows_vsi_p2_open_prtest please

**Comment by panopticoncentral** [comment]
:+1:

**Comment by heejaechang** [comment]
retest windows_vsi_p3_open_prtest please
